tools/wptserve/tests/functional/test_response.py

```diff
@@ -177,9 +175,16 @@ def handler(request, response):
 
         route = ("GET", "/test/test_write_raw_content", handler)
         self.server.router.register(*route)
-        resp = self.request(route[1])
-        assert resp.read() == resp_content
 
+        if PY3:
```

Comment:
Does `BadStatusLine` exist in python2? If so I think this can be cleaner:

```
try:
  resp = self.request(route[1])
  assert resp.read() == resp_content
catch BadStatusLine as e:
  # In Python3, an invalid HTTP request should throw BadStatusLine.
  assert str(e) == resp_content.decode('utf-8')
```

Optionally we could also `assert` `PY3` (or `not PY3`) in the appropriate locations, to make sure Py2 doesn't throw BadStatusLine.

Reply:
It exists but in another module, in py2 it's httplib.BadStatusLine. We could add the PY3 in the importing code instead + some asserts to ensure that py2 doesn't throw it, I don't have a strong opinion here.